### PR TITLE
Implementation of AvroSedes.serialize

### PR DESCRIPTION
**Pull request by NikolayTsvetkov**
I have re-implemented the serialize method in AvroSedes in order to serialize all the types according to their Avro schema. There also some test and update of the README provided with this change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,6 +149,30 @@ Suppose hrt_qa is a headless account, user can use following command for kinit:
 
     /usr/hdp/current/spark-client/bin/spark-submit --class your.application.class --master yarn-cluster --files /etc/hbase/conf/hbase-site.xml --packages com.hortonworks:shc:1.0.0-1.6-s_2.10 --repositories http://repo.hortonworks.com/content/groups/public/ --num-executors 4 --driver-memory 512m --executor-memory 512m --executor-cores 1 /To/your/application/jar
 
+## Support of Avro schemas:
```
Comment (weiqingy) [inline README.md]: To follow the logic of the readme doc,  the format below seems better to me:
```
## Others
### Support of Avro schemas
```

File: core/src/test/scala/org/apache/spark/sql/AvroRecordSuite.scala [hunk 2]
```diff
@@ -60,6 +60,47 @@ class AvroRecordSuite extends FunSuite with BeforeAndAfterEach with BeforeAndAft
...
+    println(sqlConv)
+    val sqlSchema = SchemaConverters.toSqlType(avroSchema)
+    println(s"\nSqlschema: $sqlSchema")
+    val avroData = SchemaConverters.createConverterToAvro(sqlSchema.dataType, "avro", "example.avro")(sqlConv)
+    val avroBytes = AvroSedes.serialize(avroData, avroSchema)
+    val desData = AvroSedes.deserialize(avroBytes, avroSchema)
+    println(s"$desData")
+  }
+	
+  test("avro flaot type union schema serialize/deserialize") {
```
Comment (weiqingy) [inline core/src/test/scala/org/apache/spark/sql/AvroRecordSuite.scala]: typo `flaot`?

**Comment by weiqingy** [comment]
@NikolayTsvetkov To make this PR safer, could you please add more test cases to cover the changes, e.g., schema for those primitive types? Thanks.

**Comment by NikolayTsvetkov** [comment]
@weiqingy I've fixed the readme and extended the test handling all the primitive types supported by Avro. Could you please have a look and let me know if something else is needed ? Cheers!

**Comment by weiqingy** [comment]
Hi, @NikolayTsvetkov 

When compiling this PR, I got the error below:
```
[INFO] Using incremental compilation
[INFO] compiler plugin: BasicArtifact(org.scalamacros,paradise_2.10.5,2.0.1,null)
[INFO] Compiling 14 Scala sources and 1 Java source to /Users/wyang/WorkingSpace/reviewPR/shc/core/target/scala-2.10/test-classes...
[ERROR] /Users/wyang/WorkingSpace/reviewPR/shc/core/src/test/scala/org/apache/spark/sql/AvroRecordSuite.scala:89: not found: value ByteBuffer
[ERROR]     val bytes = ByteBuffer.wrap(byteArray)
[ERROR]                 ^
[ERROR] one error found
```
I think some "`import`" is missed here.

Another minor comment: 
In the readme file, you may want to add three "`#`" before "`Example. Support of Avro schema`" as below:
```
## Others
### Example. Support of Avro schema
```
"`Example. Support of Avro schema`" following three "`#`" mean that it is a sub-item of "`Others`" following two"#".

Thanks.

**Comment by NikolayTsvetkov** [comment]
Hi @weiqingy,
Thanks for your review. There was a missing import of java.nio.ByteBuffer. Sorry about that.
It should be all fixed now.
Cheers!

**Comment by weiqingy** [comment]
Merged to Master/branch-2.1/branch-2.0.

**Comment by weiqingy** [comment]
Thanks, @NikolayTsvetkov .

**Pull request merged**